app/scripts/models/auth_brokers/pairing/supplicant.js

```diff
@@ -0,0 +1,59 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import OAuthRedirectBroker from '../oauth-redirect';
+import ChannelServerClient from '../../../lib/channel-server-client';
+import setRemoteMetaData from './remote-metadata';
+import SupplicantStateMachine from '../../pairing/supplicant-state-machine';
+import Url from '../../../lib/url';
+
+export default class SupplicantBroker extends OAuthRedirectBroker {
+  type = 'supplicant';
+
+  initialize (options = {}) {
+    super.initialize(options);
+
+    const { config, notifier, relier } = options;
+    const channelServerUri = config.pairingChannelServerUri;
+    const { channelId, channelKey } = relier.toJSON();
+    if (channelId && channelKey && channelServerUri) {
+      this.channelServerClient = new ChannelServerClient({
+        channelId,
+        channelKey,
+        channelServerUri,
+      },
+      {
+        notifier,
+      });
+
+      this.suppStateMachine = new SupplicantStateMachine({}, {
+        broker: this,
+        channelServerClient: this.channelServerClient,
+        notifier,
+        relier
+      });
+
+      this.channelServerClient.open();
+    }
+  }
+
+  afterSupplicantApprove () {
+    return Promise.resolve().then(() => {
+      this.notifier.trigger('pair:supp:authorize');
+    });
+  }
+
+  sendCodeToRelier () {
+    return Promise.resolve().then(() => {
+      const relier = this.relier;
+      const result = {
+        redirect: Url.updateSearchString(relier.get('redirectUri'), relier.pick('code', 'state'))
```

Comment:
Removed that extra check for now, I think it was a leftover? and after that the combining didn't make sense because the return points of the 2 broker methods are different,.